sycl/source/detail/kernel_bundle_impl.hpp

```diff
@@ -220,35 +220,173 @@ class kernel_bundle_impl {
     // TODO: Unify with c'tor for sycl::compile and sycl::build by calling
     // sycl::join on vector of kernel_bundles
 
-    // The loop below just links each device image separately, not linking any
-    // two device images together. This is correct so long as each device image
-    // has no unresolved symbols. That's the case when device images are created
-    // from generic SYCL APIs. There's no way in generic SYCL to create a kernel
-    // which references an undefined symbol. If we decide in the future to allow
-    // a backend interop API to create a "sycl::kernel_bundle" that references
-    // undefined symbols, then the logic in this loop will need to be changed.
+    // Due to a bug in L0, specializations with conflicting IDs will overwrite
+    // each other when linked together, so to avoid this issue we link
+    // regular offline-compiled SYCL device images in separation.
+    // TODO: Remove when spec const overwriting issue has been fixed in L0.
+    std::vector<const DevImgPlainWithDeps *> OfflineDeviceImages;
+    std::set<std::shared_ptr<device_image_impl>> OfflineDeviceImageSet;
```

Comment:
Same question about ordered containers in this file (except for heterogenous lookup cases).